### PR TITLE
Pdp module

### PR DESCRIPTION
**Pull request by zahinSJI**
PDP module automation complete

Report Link: https://drive.google.com/file/d/17_-byKxHX0dEZMe4UGBNEqottYFDqsui/view?usp=sharing

### CODE REVIEW

File: BenadrylAutomation/Configurations/config.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
-stage_url=http://test.benadryl.com/
+stage_url=http://app-23191.on-aptible.com/
 live_url=https://www.benadryl.com/
 browserName=Chrome
-headlessChrome=yes
+headlessChrome=No
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/Configurations/config.properties]: @zahinSJI why this value is set no??? It should be set always yes before creating PR

File: BenadrylAutomation/Configurations/config.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
-stage_url=http://test.benadryl.com/
+stage_url=http://app-23191.on-aptible.com/
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/Configurations/config.properties]: @zahinSJI what are you doing man? this url is from other project. Why did u changed this ? and added here??

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java [hunk 3]
```diff
@@ -1004,30 +1004,16 @@ public static void navigateBack() {
 	public static void navigateForward() {
 		driver.navigate().forward();
 	}
-
+	
 	/**
 	 * Function to verify page url
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/actions/Action.java:1009]: @zahinSJI there should be empty line after this. i can see you have removed that, pls add again

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java [hunk 4]
```diff
@@ -1004,30 +1004,16 @@ public static void navigateBack() {
...
-	 * 
 	 * @param partialUrl pass partial url to verify
 	 */
 	public static void verifyPageUrl(String partialUrl) {
 		String currentUrl = driver.getCurrentUrl().trim();
-		extentInfoLog("Asserting current URL : ", currentUrl);
+        extentInfoLog("Asserting current URL : ", currentUrl);
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/actions/Action.java]: @zahinSJI format this file properly.

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java (hunk 3 above)
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/actions/Action.java]: @zahinSJI there should be empty line after this. i can see you have removed that, pls add again

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java [hunk 5]
```diff
@@ -1004,30 +1004,16 @@ public static void navigateBack() {
...
-	 * Function to hover over element using actions class
-	 * 
-	 * @param driver  pass driver object
-	 * @param element pass webelement to hover over
-	 * @throws InterruptedException
-	 */
-	public static void hoverOverElement(WebDriver driver, WebElement element) throws InterruptedException {
-		Actions action = new Actions(driver);
-		action.moveToElement(element).perform();
 	}
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/actions/Action.java]: @zahinSJI why did you removed hoverOverElement() function?? Is it creating problem??

File: BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java [hunk 6]
```diff
@@ -63,15 +58,11 @@ public void loadConfig() throws IOException {
 
 	public static void launchApplication() {
 		String browserName = prop.getProperty("browserName");
-		runOn = prop.getProperty("runOn");
-		String headlessChrome = prop.getProperty("headlessChrome");
+		String runOn = prop.getProperty("runOn");
 		if (browserName.contains("Chrome")) {
 			ChromeOptions options = new ChromeOptions();
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java]: @zahinSJI why did you removed code for headless chrome??

File: BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java [hunk 7]
```diff
@@ -37,14 +35,11 @@ public class BaseClass {
 	public static ExtentTest test;
 	public String defaultFlag = "Yes";
 	public static String baseURI;
-	public static String runOn;
 	public ExtentReports exprep = ExtentManager.setExtent();
 
 	public static HeaderPage header;
 	public static HomePage home;
 	public static DifferencesPage difference;
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java:48]: @zahinSJI i cannot see declaration of comapre and signup... why?

File: BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java [hunk 8]
```diff
@@ -81,11 +72,10 @@ public static void launchApplication() {
 		header = new HeaderPage();
 		home = new HomePage();
 		difference = new DifferencesPage();
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java:89]: @zahinSJI initialization of compare and signup is missing

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+    @FindBy(id = "mini-panel-product_faq")
+    WebElement faqsSection;
+
+    @FindBy(css = "#mini-panel-product_faq span")
+    WebElement faqsHeading;
+
+    @FindBy(css = "#bv-product-reviews")
+    WebElement reviewsSection;
+	
+	@FindBy(xpath = "//div[@id='onetrust-close-btn-container']")
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:101]: @zahinSJI format this file properly

File: BenadrylAutomation/Configurations/config.properties (hunk 2 above)
Comment (zahinSJI) [reply BenadrylAutomation/Configurations/config.properties]: image

It is absolutely correct in my end, I haven't made any change to the file other than making it headed

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+	WebElement closePrivacyBtn;
+
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:103]: @zahinSJI remove this extra empty line. and do this for other functions as well.

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+	WebElement closePrivacyBtn;
+    
+    public WebElement getJumpToHeadings(String headingText) {
+        return driver.findElement(By.xpath("//div[@class='jump-to-wrapper']//ul//li//a[text()='" + headingText + "']"));
+    }
+	
+	/**
+	 * Function to click on close privacy pop-up button
+	 */
+	public void closePrivacyPopup() {
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:111]: @zahinSJI why have you writtern this function?? its already there in home page

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java (hunk 4 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/actions/Action.java]: Format is fine in my end
image

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+        extentInfoLog("Image is displayed : ", Action.isDisplayed(driver, prodImg));
+    }
+
+    /**
+     * Visits a product page from the homepage.
+     * Then, verifies the title of the product.
+     */
+    public void verifyProdTitle() {
+    	this.visitPDP();
+        extentInfoLog("Title is displayed : ", Action.isDisplayed(driver, prodTitle));
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:143]: @zahinSJI why are you not asserting this?? why just display??

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+     * Then, verifies if the overview of the product is visible.
+     */
+    public void verifyProdOverview() {
+    	this.visitPDP();
+        extentInfoLog("Overview is displayed : ", Action.isDisplayed(driver, prodOverview));
+    }
+
+    /**
+     * Visits a product page from the homepage.
+     * Then, verifies the buttons.
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:167]: @zahinSJI param is missing

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+            this.verifyWriteAReviewBtn(expectedText);
+        } else {
+        	this.verifyBuyNowBtn(expectedText);
+        }
+    }
+
+    /**
+     * Visits a product page from the homepage.
+     * Then, verifies the Write a Review button.
+     */
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:180]: @zahin param missing. pls do it for other fuctions

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java (hunk 5 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/actions/Action.java]: Function is present in my code
image

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+        extentInfoLog("Section is displayed : ", Action.isDisplayed(driver, faqsSection));
+        Assert.assertEquals(actualHeadingText, sectionHeading);
+        extentInfoLog(jumpToHeadingText + "jump heading verified");
+    }
+
+    /**
+     * Visits a product page from the homepage.
+     * Then, verifies the jump to heading.
+     */
+    public void verifyJumpToReview(String jumpToHeading) {
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:305]: @zahinSJI what is this jumpTo??? write a meaningful name for the function. Everywhere you have used jump

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+     */
+    public ProductPage() {
+        PageFactory.initElements(driver, this);
+        actions = new Actions(driver);
+    }
+
+    /**
+     * Locators
+     */
+    @FindBy(css = "#tab-menu-8651")
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:32]: @zahinSJI next time try to use xpath over css selector

File: BenadrylAutomation/src/main/java/com/jnj/actions/Action.java (hunk 3 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/actions/Action.java]: There is an empty line after every new function
image

File: BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java (hunk 6 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java]: The code is still there in my code
image

File: BenadrylAutomation/src/test/java/com/jnj/testcases/DifferencePageTest.java [hunk 17]
```diff
@@ -14,59 +14,50 @@
...
 
 public class DifferencePageTest extends BaseClass {
-
 	@BeforeMethod
 	public void setup() throws InterruptedException {
 		launchApplication();
 	}
 
-	@Test(priority = 1, dataProvider = "urlAndTitle", dataProviderClass = DifferenceProvider.class)
+	@Test(priority = 1, description = "Verify URL and title", dataProvider = "urlAndTitle", dataProviderClass = DifferenceProvider.class)
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/test/java/com/jnj/testcases/DifferencePageTest.java]: @zahinSJI why did you added description for this file???

File: BenadrylAutomation/Configurations/config.properties (hunk 1 above)
Comment (zahinSJI) [reply BenadrylAutomation/Configurations/config.properties]: Change applied

File: BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java (hunk 7 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java:48]: They were added and staged, they were not pushed for unknown reason.

File: BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java (hunk 8 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/base/BaseClass.java:89]: They were added and staged, they were not pushed for unknown reason.

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java (hunk 9 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:101]: Format okay from my end

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java (hunk 10 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:103]: Format okay from my end

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java (hunk 11 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:111]: I have added it in the PDP pom file to not keep any dependency with home.

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java (hunk 16 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:32]: Noted

File: BenadrylAutomation/src/test/java/com/jnj/testcases/DifferencePageTest.java (hunk 17 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/test/java/com/jnj/testcases/DifferencePageTest.java]: I did not change any other file except for PDP

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java (hunk 15 above)
Comment (zahinSJI) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/ProductPage.java:305]: It is name of a particular set of menu

File: BenadrylAutomation/Configurations/config.properties (hunk 2 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/Configurations/config.properties]: @zahinSJI can you please commit this file again?
i cannot see the change here.

File: BenadrylAutomation/Configurations/config.properties (hunk 1 above)
Comment (vaibhavNagvekarSJI) [reply BenadrylAutomation/Configurations/config.properties]: @zahinSJI cannot see the change

**Comment by vaibhavNagvekarSJI** [comment]
Master is not merged to this PR correctly hence cannot see some of the files, eighter updated code. So rejecting the PR.
@zahinSJI

**Pull request closed without merging**